scripts/core/mutableCommandList.yml

```diff
@@ -0,0 +1,269 @@
+name: $x_mutable_kernel_argument_exp_desc_t
+base: $x_base_desc_t
+members:
+    - type: uint64_t
+      name: commandId
+      desc: "[in] command identifier"
+    - type: uint32_t
+      name: argIndex
+      desc: "[in] kernel argument index"
+    - type: size_t
+      name: argSize
+      desc: "[in] kernel argument size"
+    - type: "const void*"
+      name: pArgValue
+      desc: "[in] pointer to kernel argument value"
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Mutable kernel group count descriptor"
+version: "1.9"
+class: $xCommandList
+name: $x_mutable_group_count_exp_desc_t
+base: $x_base_desc_t
+members:
+    - type: uint64_t
+      name: commandId
+      desc: "[in] command identifier"
+    - type: "const $x_group_count_t*"
+      name: pGroupCount
+      desc: "[in] pointer to group count"
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Mutable kernel group size descriptor"
+version: "1.9"
+class: $xCommandList
+name: $x_mutable_group_size_exp_desc_t
+base: $x_base_desc_t
+members:
+    - type: uint64_t
+      name: commandId
+      desc: "[in] command identifier"
+    - type: uint32_t
+      name: groupSizeX
+      desc: "[in] group size for X dimension to use for the kernel"
+    - type: uint32_t
+      name: groupSizeY
+      desc: "[in] group size for Y dimension to use for the kernel"
+    - type: uint32_t
+      name: groupSizeZ
+      desc: "[in] group size for Z dimension to use for the kernel"
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Mutable kernel global offset descriptor"
+version: "1.9"
+class: $xCommandList
+name: $x_mutable_global_offset_exp_desc_t
+base: $x_base_desc_t
+members:
+    - type: uint64_t
+      name: commandId
+      desc: "[in] command identifier"
+    - type: uint32_t
+      name: offsetX
+      desc: "[in] global offset for X dimension to use for this kernel"
+    - type: uint32_t
+      name: offsetY
+      desc: "[in] global offset for Y dimension to use for this kernel"
+    - type: uint32_t
+      name: offsetZ
+      desc: "[in] global offset for Z dimension to use for this kernel"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Returns a unique command identifier for the next command to be appended to a command list."
+version: "1.9"
+class: $xCommandList
+name: GetNextCommandIdExp
+details:
+    - "This function may only be called for a mutable command list."
+    - "This function may not be called on a closed command list."
+    - "This function may be called from simultaneous threads with the same command list handle."
+    - "The implementation of this function should be lock-free."
+params:
+    - type: $x_command_list_handle_t
+      name: hCommandList
+      desc: "[in] handle of the command list"
+    - type: "const $x_mutable_command_id_exp_desc_t*"
```

Comment:
part of `x_mutable_command_id_exp_desc_t` is `x_mutable_command_list_exp_flags_t` - does it mean we can specify in advance when getting CommandID which part of next append are going to mutate?

Reply:
ok, thanks

Reply:
@zzdanowicz The mutable command id descriptor does not currently contain the command list flags, only the command flags. The command list _properties_ reports out both command list and command flags.

Is it a requirement that the command _list_ flags be built into the command id descriptor? The command flags should be sufficient.

Reply:
Yes, *command* flags. I just added (embargo) definitions, for the development purposes before official header arrive.
Treat them as temporary and only for sake of development - did I got it right?
```
typedef uint32_t ze_mutable_command_exp_flags_t;
typedef enum _ze_mutable_command_exp_flag_t {
    ZE_MUTABLE_COMMAND_EXP_FLAG_KERNEL_ARGUMENTS = ZE_BIT(0),
    ZE_MUTABLE_COMMAND_EXP_FLAG_GROUP_COUNT = ZE_BIT(1),
    ZE_MUTABLE_COMMAND_EXP_FLAG_GROUP_SIZE = ZE_BIT(2),
    ZE_MUTABLE_COMMAND_EXP_FLAG_GLOBAL_OFFSET = ZE_BIT(3),
    ZE_MUTABLE_COMMAND_EXP_FLAG_SIGNAL_EVENT = ZE_BIT(4),
    ZE_MUTABLE_COMMAND_EXP_FLAG_WAIT_EVENTS = ZE_BIT(5),
    ZE_MUTABLE_COMMAND_EXP_FLAG_FORCE_UINT32 = 0x7fffffff
} ze_mutable_command_exp_flag_t;

#define ZE_STRUCTURE_TYPE_MUTABLE_COMMAND_ID_EXP_DESC (ze_structure_type_t)0x00040001

typedef struct _ze_mutable_command_id_exp_desc_t {
    ze_structure_type_t stype;
    const void *pNext;
    ze_mutable_command_exp_flags_t flags;
} ze_mutable_command_id_exp_desc_t;
```